### PR TITLE
opencv test draft for ImageContainer

### PR DESCRIPTION
**Pull request by markram1729**
Draft PR for opencv test integration #115

### CODE REVIEW

File: tests/Interface/core/ImageContainerTest.cpp
Comment (meshtag) [inline tests/Interface/core/ImageContainerTest.cpp:36]: Can you use a more descriptive name here?

File: tests/Interface/core/ImageContainerTest.cpp
Comment (meshtag) [inline tests/Interface/core/ImageContainerTest.cpp:54]: No need to explicitly return false here, just return the function result directly.

File: tests/Interface/core/ImageContainerTest.cpp
Comment (meshtag) [inline tests/Interface/core/ImageContainerTest.cpp:58]: Both of these branches (if and else block) can be simplified using a ternary operator.

File: tests/Interface/core/ImageContainerTest.cpp
Comment (meshtag) [inline tests/Interface/core/ImageContainerTest.cpp:47]: Do you really require signedness from the variable here? You can use unsigned in all cases where it is not required.

File: tests/Interface/core/ImageContainerTest.cpp
Comment (meshtag) [inline tests/Interface/core/ImageContainerTest.cpp:31]: You can use `std::abs` to eliminate an unnecessary branch here.

File: tests/Interface/core/ImageContainerTest.cpp
Comment (meshtag) [inline tests/Interface/core/ImageContainerTest.cpp:32]: No need for explicit true and false here, you can just return the comparison result itself.

**Review by meshtag — CHANGES_REQUESTED** [review]
Thanks for the work! 

I have added some high level comments.
Additionally, 
1. Can you please format everything with clang-format.
2. It is useful to have a descriptive summary of what the function intends to do on top of its definition/declaration.
3. It is a good idea to separate logically different lines of code with a newline for better readability.

File: tests/Interface/core/ImageContainerTest.cpp
Comment (meshtag) [inline tests/Interface/core/ImageContainerTest.cpp:28]: No need to have this separate control flow branch here. Please handle it generically with below logic itself.

File: tests/Interface/core/ImageContainerTest.cpp
Comment (meshtag) [inline tests/Interface/core/ImageContainerTest.cpp:30]: No need to store the diff in a separate variable here.

File: tests/Interface/core/ImageContainerTest.cpp
Comment (meshtag) [inline tests/Interface/core/ImageContainerTest.cpp:39]: Nit: Using pre-increment (++i instead of i++) is marginally better wherever you can (pls modify other places as well).

File: tests/Interface/core/ImageContainerTest.cpp
Comment (meshtag) [inline tests/Interface/core/ImageContainerTest.cpp:48]: Extra set of parentheses here?

File: tests/Interface/core/ImageContainerTest.cpp
Comment (meshtag) [inline tests/Interface/core/ImageContainerTest.cpp:174]: Did you format this file with clang-format?

`git clang-format HEAD~` will format your entire previous commit in-place for you.

File: tests/Interface/core/ImageContainerTest.cpp
Comment (meshtag) [inline tests/Interface/core/ImageContainerTest.cpp:46]: These loops have redundant braces.

File: tests/Interface/core/ImageContainerTest.cpp
Comment (meshtag) [inline tests/Interface/core/ImageContainerTest.cpp:44]: Can you make this more generic instead of only handling for 2 dimensions?

**Review by meshtag — COMMENTED** [review]
This is looking good. I have added few more comments.

File: tests/Interface/core/ImageContainerTest.cpp
Comment (markram1729) [reply tests/Interface/core/ImageContainerTest.cpp:41]: can you more give more information , generic as in for N>3 
but as i see at Imagecontainer.h ,for (N!=2)  cv image is not handled when N==3 / for NCHW layout i see copying only first channel information of image than number of channels 

can you give me direction /help on what can be done

Comment (meshtag) [reply tests/Interface/core/ImageContainerTest.cpp:41]: Can you support testing with the NCHW layout? (and add a TODO here to note that tests should be updated there once more layouts are supported functionally)

Comment (markram1729) [reply tests/Interface/core/ImageContainerTest.cpp:41]: can you please check now

**Comment by markram1729** [comment]
Can @meshtag  @zhanghb97  review my pull request, waiting for your suggestions

File: tests/Interface/core/ImageContainerTest.cpp
Comment (meshtag) [inline tests/Interface/core/ImageContainerTest.cpp:473]: These tests are testing the same thing which was tested by above tests? If not, please add a comment explaining that.

**Comment by meshtag** [comment]
> Can @meshtag @zhanghb97 review my pull request, waiting for your suggestions

Sorry, I missed this. Can you please format all files with clang-format. 

Commented a minor suggestion, rest looks fine to me. 

Please wait for the approval of the person maintaining this part currently. Thanks!

File: tests/Interface/core/ImageContainerTest.cpp
Comment (markram1729) [reply tests/Interface/core/ImageContainerTest.cpp:466]: yes , above tests checks only on single images , here i have batched two images into a blob that is the only difference 

and i have added few required  segments in ImageContainer.h for batched images

**Comment by zhanghb97** [comment]
Thanks!

**Pull request merged**
